addons/mail/models/discuss/discuss_channel.py

```diff
@@ -271,33 +271,18 @@ def write(self, vals):
             failing_channels = self.filtered(lambda channel: channel.channel_type != vals.get('channel_type'))
             if failing_channels:
                 raise UserError(_('Cannot change the channel type of: %(channel_names)s', channel_names=', '.join(failing_channels.mapped('name'))))
+        old_vals = {channel.id: {key: channel[key] for key in vals.keys()} for channel in self}
+        result = super().write(vals)
         notifications = []
         for channel in self:
-            current_val = channel.read(vals.keys())[0]
-            diff = {}
-            for key in vals.keys():
-                if current_val.get(key) != vals.get(key) and key != "image_128":
-                    diff[key] = vals[key]
-            if diff:
-                notifications.append([channel, "mail.record/insert", {
-                    "Thread": {
-                        "id": current_val["id"],
-                        "model": "discuss.channel",
-                        **diff
-                    }
-                }])
-        result = super().write(vals)
+            current_val = {key: channel[key] for key in vals.keys()}
+            info = channel._channel_basic_info()
+            for key, value in current_val.items():
+                if (key in info or key == "image_128") and value != old_vals.get(channel.id)[key]:
```

Comment:
keys in info are transformed so it won't work as expected either, the "observed" keys should probably be simply listed instead